### PR TITLE
fix RemoteChannel tests

### PR DESCRIPTION
**Pull request by amitmurthy**
@timholy , please have a look.

### CODE REVIEW

**Comment by timholy** [comment]
I'm an infrequent diver into the world of parallel, so I haven't followed exactly what these are and what these settings should be. I trust you implicitly in all such matters :smile:, and thanks for fixing!

**Comment by amitmurthy** [comment]
😄 . Likewise for me and `Compat`. Therefore, just wanted a quick review before merging.

File: test/runtests.jl [hunk 1]
```diff
@@ -1003,10 +1003,12 @@ rm(dirwalk, recursive=true)
 r = remotecall(sin, 1, pi/3)
 @compat foo(r::Future) = 7
 @test foo(r) == 7
-if VERSION < v"0.4.0"
-    @compat rc = RemoteChannel(1,2,3)
-else
-    @compat rc = RemoteChannel{Channel{Any}}(1,2,3)
+if VERSION < v"0.5.0-dev+1343"
```
Comment (martinholters) [inline test/runtests.jl]: Why are these tests not executed for current Julia?

Comment (amitmurthy) [inline test/runtests.jl]: Yes, they should be - will push an update.

**Comment by amitmurthy** [comment]
Good to merge? The error seems unrelated.

**Comment by hayd** [comment]
It does seem unrelated, and the tests certainly fail without this.

**Pull request merged**